rclpy/rclpy/node.py

```diff
@@ -999,6 +999,9 @@ def _set_parameters_atomically_common(
         elif self._on_set_parameters_callbacks:
             for callback in self._on_set_parameters_callbacks:
                 result = callback(parameter_list)
+                if result is None:
+                    warnings.warn('Callback returned None, it should return SetParameterResult.')
+                    result = SetParametersResult(successful=False, reason='Callback returned None')
```

Comment:
Apart from the cases where None is returned, should we check if the returned result is not of the SetParametersResult type?  Because if it is not of the SetParametersResult type, the subsequent code `result.successful` will also cause an error.

Reply:
that type is `Any`. or are you suggesting we should have `SetParameterResult` as python class object?

Reply:
Ah sorry, i misunderstood. just checking if `SetParametersResult` instance?

Reply:
fixed in https://github.com/ros2/rclpy/pull/1425/commits/ed4aa6df8fc5ce185bee486d8c7868653e58f6e0